featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -927,3 +939,7 @@ def _features_have_same_path(input_features):
 def _direct_of_entity(feature, parent_entity):
     return isinstance(feature, DirectFeature) \
         and feature.parent_entity.id == parent_entity.id
+
+
+def _sort_primitives(prim_list):
+    return sorted(prim_list, key=lambda prim: prim if isinstance(prim, str) else prim.name)
```

Comment:
Oh yeah, this wouldn't handle that case.

I changed it so that we're also using `get_args_string` to sort when it's applicable and added `Mean(skipna=False), Mean` to the test to catch that case